common/txmgr/broadcaster.go

```diff
@@ -412,10 +463,19 @@ func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) Sync
 					}
 					continue
 				}
-				return
+				break loop
```

Comment:
I _think_ there is a simpler way to write this loop with a closure that would dedupe some logs as well, but more importantly, does this loop yield to a cancelled context? Or would it loop forever - blocking the node from shutting down gracefully?

Reply:
I refactored this to avoid the break tag. I believe a cancelled context would stop the loop in the existing code. There's a switch case that returns out of the function if a signal is detected on the broadcaster's chStop channel